### PR TITLE
Remove Java's Http.Context and JPAApi thread-locals

### PR DESCRIPTION
**Pull request by mkurz**
Time to say goodbye to `Http.Context`, `Http.Response` and `JPAEntityManagerContext` :stuck_out_tongue_winking_eye:

```bash
./framework$ sbt compile
```
works already. Therefore we can be sure we didn't forget to deprecate anything related.
With this pull request Java's `Session` and `Flash` also finally become immutable.

Missing:
- [x] Fix tests which still make us of deprecated (and now removed) classes
- [x] Remove `play.allowHttpContext` and `play.jpa.allowJPAEntityManagerContext`, see [docs](https://www.playframework.com/documentation/2.7.x/JavaHttpContextMigration27#Disabling-the-Http.Context-and-JPA-thread-local)

### CODE REVIEW

**Comment by dwijnand** [comment]
I've started exploring this in a branch too.  When I get back to it I'll compare with your changes here.

**Comment by dwijnand** [comment]
(I mean just to highlight different decisions, I'm happy for you to complete the work.)

**Comment by mkurz** [comment]
@dwijnand :+1: Back when I was working on this pull request, it wasn't finished yet. Tests weren't fixed, etc. The point of this pull request was to see if everything compiles (or if there are any warnings) when removing `Http.Context` and other thread locals.
Also, as you can see there are still many conflicts because this was done before the code restructuring.

**Comment by mkurz** [comment]
Hey @marcospereira, why did you remove this from the Play 2.8 milestone? I was thinking about working on this soonish.

**Comment by dwijnand** [comment]
It was meant to be in the 2.x milestone.  But note that it just means that we're not planning on making this change in 2.8, but that doesn't mean that it can't happen in 2.8.

TL;DR we'll merge this if it's ready 😄

**Comment by mkurz** [comment]
@dwijnand Got it :+1:

**Comment by mkurz** [comment]
I will split this pull request into several smaller ones, starting with #9269.

**Comment by ignasi35** [comment]
> I will split this pull request into several smaller ones, starting with #9269.

Shall we close this PR, then?

**Comment by mkurz** [comment]
I may force push the final changes to this pull request. Let's keep it open, I will work on this in the coming days anyway.

**Comment by mkurz** [comment]
Superseded by #9320
